query/engine.go

```diff
@@ -25,16 +25,30 @@ type LocalEngine struct {
 	tableProvider logicalplan.TableProvider
 }
 
+type Option func(*LocalEngine)
+
+func WithTracer(tracer trace.Tracer) Option {
+	return func(e *LocalEngine) {
+		e.tracer = tracer
+	}
+}
+
 func NewEngine(
 	pool memory.Allocator,
```

Comment:
I always wonder if we should do the same with these `memory.Allocator`.
Totally out of scope for this PR, maybe creating an issue for outside contributions. That would be a nice one, I imagine.

Reply:
Totally agree, I strongly considered removing it in this PR as well but decided to leave it for now